presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -696,4 +731,285 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
                 .map(id -> IcebergUtil.resolveSnapshotId(table, id))
                 .or(() -> Optional.ofNullable(table.currentSnapshot()).map(Snapshot::snapshotId)));
     }
+
+    @Override
+    public void createMaterializedView(ConnectorSession session, SchemaTableName viewName, ConnectorMaterializedViewDefinition definition, boolean replace, boolean ignoreExisting)
+    {
+        HiveIdentity identity = new HiveIdentity(session);
+        Optional<Table> existing = metastore.getTable(identity, viewName.getSchemaName(), viewName.getTableName());
+
+        // It's a create command where the materialized view already exists and 'if not exists' clause is not specified
+        if (!replace && existing.isPresent() && !ignoreExisting) {
+            throw new MaterializedViewAlreadyExistsException(viewName);
+        }
+
+        // Generate a storage table name and create a storage table. The properties in the definition are table properties for the
+        // storage table as indicated in the materialized view definition.
+        String storageTableName = "st_" + UUID.randomUUID().toString().replace("-", "");
+        Map<String, Object> storageTableProperties = new HashMap<>(definition.getProperties());
+        if (!storageTableProperties.containsKey(FILE_FORMAT_PROPERTY)) {
+            storageTableProperties.put(FILE_FORMAT_PROPERTY, DEFAULT_FILE_FORMAT_DEFAULT);
+        }
+
+        SchemaTableName storageTable = new SchemaTableName(viewName.getSchemaName(), storageTableName);
+        List<ColumnMetadata> columns = new ArrayList<>();
+        for (ConnectorMaterializedViewDefinition.Column col : definition.getColumns()) {
+            columns.add(new ColumnMetadata(col.getName(), typeManager.getType(col.getType())));
+        }
+        ConnectorTableMetadata tableMetadata = new ConnectorTableMetadata(storageTable, columns, storageTableProperties, Optional.empty());
+        Optional<ConnectorNewTableLayout> layout = getNewTableLayout(session, tableMetadata);
+        finishCreateTable(session, beginCreateTable(session, tableMetadata, layout), ImmutableList.of(), ImmutableList.of());
+
+        // Create a view indicating the storage table
+        ImmutableMap<String, String> viewProperties = ImmutableMap.<String, String>builder()
+                .put(PRESTO_QUERY_ID_NAME, session.getQueryId())
+                .put(STORAGE_TABLE, storageTableName)
+                .put(PRESTO_VIEW_FLAG, "true")
```

Comment:
All views recognized by presto as views excluding the hive views (that get translated) need the PRESTO_VIEW_FLAG. 
This code sets the flag on the view object. 
When we try to resolve the catalog for a materialized view, `getRawSystemTable` in IcebergMetadata gets called (we had an offline discussion about this). `getRawSystemTable` in turn calls `metastore.getTable`, which calls `ThriftHiveMetastore.getTable`. This function has code: 
`if (!translateHiveViews && table.getTableType().equals(TableType.VIRTUAL_VIEW.name()) && !isPrestoView(table)) {
                            throw new HiveViewNotSupportedException(new SchemaTableName(databaseName, tableName));
                        }`
The materialized view-view object is expected to have the PRESTO_VIEW_FLAG, otherwise it throws the `Hive views are not supported` exception.